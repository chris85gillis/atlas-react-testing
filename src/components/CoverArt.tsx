import React from 'react';

type CoverArtProps = {
  artwork?: string;
};

const CoverArt: React.FC<CoverArtProps> = ({ artwork }) => {
  return (
    <div className="flex justify-center items-center mt-6">
      <img
        src={artwork}
        alt="Cover Art"
        className="w-[25rem] h-[22rem] object-cover rounded-lg shadow-custom"
      />
    </div>
  );
};

export default CoverArt;