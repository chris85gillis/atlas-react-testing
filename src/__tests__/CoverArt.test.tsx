import { render } from '@testing-library/react';
import CoverArt from '../components/CoverArt';

describe('CoverArt', () => {
  it('renders correctly with a valid artwork URL', () => {
    const { asFragment } = render(<CoverArt artwork="https://example.com/artwork.jpg" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with an empty artwork URL', () => {
    const { asFragment } = render(<CoverArt artwork="" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with a missing artwork prop', () => {
    const { asFragment } = render(<CoverArt artwork={undefined} />);
    expect(asFragment()).toMatchSnapshot();
  });
});