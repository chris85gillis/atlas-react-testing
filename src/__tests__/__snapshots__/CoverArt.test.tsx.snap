// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt > renders correctly with a missing artwork prop 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center mt-6"
  >
    <img
      alt="Cover Art"
      class="w-[25rem] h-[22rem] object-cover rounded-lg shadow-custom"
    />
  </div>
</DocumentFragment>
`;

exports[`CoverArt > renders correctly with a valid artwork URL 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center mt-6"
  >
    <img
      alt="Cover Art"
      class="w-[25rem] h-[22rem] object-cover rounded-lg shadow-custom"
      src="https://example.com/artwork.jpg"
    />
  </div>
</DocumentFragment>
`;

exports[`CoverArt > renders correctly with an empty artwork URL 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center mt-6"
  >
    <img
      alt="Cover Art"
      class="w-[25rem] h-[22rem] object-cover rounded-lg shadow-custom"
      src=""
    />
  </div>
</DocumentFragment>
`;
